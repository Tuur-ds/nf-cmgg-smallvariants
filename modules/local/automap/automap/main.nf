process AUTOMAP_AUTOMAP {
    tag "$meta.id"
    label 'process_single'

    container "cmgg/automap:1.0.0"

    input:
    tuple val(meta), path(vcf), path(tbi)
    tuple val(meta2), path(repeats)
    tuple val(meta3), path(panel)
    val(genome)

    output:
    tuple val(meta), path("${prefix}"), emit: automap
    path  "versions.yml"              , emit: versions

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"

    def panel_file = panel ? "--panel $panel" : "--panel /usr/local/lib/automap/Resources/Biomodule_20220808_all_genes_hg38.txt"
    def hg_genome = genome ?: "hg38"
    def VERSION = "1.0.0"

    """
    automap \\
        --vcf $vcf \\
        --genome $hg_genome \\
        --out $prefix/ \\
        --repeats $repeats \\
        $panel_file \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        automap: $VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def panel_name = args.contains("--panelname") ? args.split("--panelname")[-1].trim().split(" ")[0] : ""
    prefix = task.ext.prefix ?: "${meta.id}"

    def create_outputs = meta.family_count > 1 ? (1..meta.family_count).collect { number ->
        def cmd_prefix = "touch ${prefix}/sample${number}"
        [
            "mkdir ${prefix}/sample${number}",
            "${cmd_prefix}/sample${number}.HomRegions.pdf",
            "${cmd_prefix}/sample${number}.HomRegions.${panel_name}.tsv",
            "${cmd_prefix}/sample${number}.HomRegions.tsv",
            "${cmd_prefix}/sample${number}.HomRegions.strict.${panel_name}.tsv"
        ].join(" && ")
    }.join(" && ") : [
        "touch ${prefix}/${meta.id}.HomRegions.pdf",
        "touch ${prefix}/${meta.id}.HomRegions.${panel_name}.tsv",
        "touch ${prefix}/${meta.id}.HomRegions.tsv",
        "touch ${prefix}/${meta.id}.HomRegions.strict.${panel_name}.tsv"
    ].join(" && ")

    def VERSION = "1.0.0"
    """
    mkdir $prefix
    ${create_outputs}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        automap: $VERSION
    END_VERSIONS
    """
}
