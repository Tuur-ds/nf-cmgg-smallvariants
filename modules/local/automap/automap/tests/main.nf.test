nextflow_process {

    name "Test Process AUTOMAP_AUTOMAP"
    script "../main.nf"
    process "AUTOMAP_AUTOMAP"

    tag "modules"
    tag "modules_local"
    tag "automap"
    tag "automap/automap"

    setup {
        run("AUTOMAP_REPEATS") {
            script "../../repeats/main.nf"
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    "hg38"
                ]
                """
            }
        }
    }

    test("homo_sapiens - vcf, tbi, repeats, [], hg38") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', family_samples:"NA24143,NA24149,NA24385", caller:"haplotypecaller" ], // meta map
                    file(params.famvcf, checkIfExists: true),
                    file(params.famtbi, checkIfExists: true)
                ]
                input[1] = AUTOMAP_REPEATS.out.repeats
                input[2] = [[],[]]
                input[3] = "hg38"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.automap.collect { meta, out ->
                        [
                            meta,
                            path(out).list().collect {
                                it.list().collect {
                                    "${file(it.toString()).name},variantsMD5:${it.vcf.variantsMD5}"
                                }
                            }
                        ]
                    },
                    process.out.versions
                ).match() }
            )
        }

    }

    test("homo_sapiens - vcf, tbi, repeats, [], hg38 - stub") {

        options "-stub"
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', family_samples:"NA24143,NA24149,NA24385", caller:"haplotypecaller" ], // meta map
                    file(params.famvcf, checkIfExists: true),
                    file(params.famtbi, checkIfExists: true)
                ]
                input[1] = AUTOMAP_REPEATS.out.repeats
                input[2] = [[],[]]
                input[3] = "hg38"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
