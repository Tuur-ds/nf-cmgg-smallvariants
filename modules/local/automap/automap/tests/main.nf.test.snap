{
    "homo_sapiens - vcf, tbi, repeats, [], hg38": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "family_samples": "NA24143,NA24149,NA24385",
                        "caller": "haplotypecaller"
                    },
                    [
                        [
                            "NA24143.individual.vcf,variantsMD5:d87c6d5aea196c746312a8a383eede4b"
                        ],
                        [
                            "NA24149.individual.vcf,variantsMD5:24eacd6958a767b7768b4282caea52cc"
                        ],
                        [
                            "NA24385.individual.vcf,variantsMD5:f7f27a20139824e3e49cf17a2867489b"
                        ]
                    ]
                ]
            ],
            [
                "versions.yml:md5,8d88bdf05fc6f578d81fee3716f8b562"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T17:03:29.44273724"
    },
    "homo_sapiens - vcf, tbi, repeats, [], hg38 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "family_samples": "NA24143,NA24149,NA24385",
                            "caller": "haplotypecaller"
                        },
                        [
                            [
                                "sample1.HomRegions.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample1.HomRegions.pdf:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample1.HomRegions.strict.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample1.HomRegions.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            [
                                "sample2.HomRegions.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample2.HomRegions.pdf:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample2.HomRegions.strict.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample2.HomRegions.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            [
                                "sample3.HomRegions.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample3.HomRegions.pdf:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample3.HomRegions.strict.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample3.HomRegions.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ]
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,8d88bdf05fc6f578d81fee3716f8b562"
                ],
                "automap": [
                    [
                        {
                            "id": "test",
                            "family_samples": "NA24143,NA24149,NA24385",
                            "caller": "haplotypecaller"
                        },
                        [
                            [
                                "sample1.HomRegions.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample1.HomRegions.pdf:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample1.HomRegions.strict.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample1.HomRegions.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            [
                                "sample2.HomRegions.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample2.HomRegions.pdf:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample2.HomRegions.strict.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample2.HomRegions.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            [
                                "sample3.HomRegions.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample3.HomRegions.pdf:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample3.HomRegions.strict.cmgg_bio.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sample3.HomRegions.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8d88bdf05fc6f578d81fee3716f8b562"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T17:00:28.503010269"
    }
}